doc/devel/contributing.rst

```diff
@@ -0,0 +1,422 @@
+      with::
+      
+        pip install coverage
+        python tests.py --with-coverage
+
+    * No pyflakes warnings, check with::
+      
+        pip install pyflakes
+        pyflakes path/to/module.py
+
+.. note::
+
+    The current state of the matplotlib code base is not compliant with all
+    of those guidelines, but we expect that enforcing those constraints on all
+    new contributions will get the overall code base quality in the right
+    direction.
+
+
+.. seealso::
+
+  * :ref:`coding_guidelines`
+  * :ref:`testing`
+  * :ref:`documenting-matplotlib`
+
+
+
+.. _new_contributors:
+
+Issues for New Contributors
+---------------------------
+
+New contributors should look for the following tags when looking for issues.
+We strongly recommend that new contributors tackle "new-contributor-friendly"
+issues (easy, well documented issues, that do not require an understanding of
+the different submodules of matplotlib) and "Easy-fix" issues. This helps the
+contributor become familiar with the contribution workflow, and for the core
+devs to become acquainted with the contributor; besides which, we frequently
+underestimate how easy an issue is to solve!
+
+.. topic:: new-contributor-friendly
+
+    A great way to start contributing to matplotlib is to pick an item from
+    the list of
+    `new-contributor-friendly <https://github.com/matplotlib/matplotlib/labels/new-contributor-friendly>`_ 
+    in the issue tracker. Resolving these issues allow you to start
+    contributing to the project without much prior knowledge. Your assistance
+    in this area will be greatly appreciated by the more experienced
+    developers as it helps free up their time to concentrate on other issues.
+ 
+
+.. _other_ways_to_contribute:
+
+Other ways to contribute
+=========================
+
+
+Code is not the only way to contribute to matplotlib. For instance,
+documentation is also a very important part of the project and often doesn't
+get as much attention as it deserves. If you find a typo in the documentation,
+or have made improvements, do not hesitate to send an email to the mailing
+list or submit a GitHub pull request. Full documentation can be found under
+the doc/ directory.
+
+It also helps us if you spread the word: reference the project from your blog
+and articles or link to it from your website!
+
+.. _coding_guidelines:
+
+Coding guidelines
+=================
+
+New modules and files: installation
+-----------------------------------
+
+* If you have added new files or directories, or reorganized existing
+  ones, make sure the new files included in the match patterns in
+  :file:`MANIFEST.in`, and/or in `package_data` in `setup.py`.
+
+C/C++ extensions
+----------------
+
+* Extensions may be written in C or C++.
+
+* Code style should conform to PEP7 (understanding that PEP7 doesn't
+  address C++, but most of its admonitions still apply).
+
+* Interfacing with Python may be done either with the raw Python/C API
+  or Cython.
```

Comment:
It might need to go back in eventually; I didn't realize it was just moved from elsewhere in our docs.  That suggests that the inclusion of cython was deliberate.  I know we talked about it, but I didn't remember having reached a conclusion.